subprojects/core/src/main/java/org/gradle/configuration/DefaultScriptPluginFactory.java

```diff
@@ -124,7 +130,7 @@ public void apply(final Object target) {
                     return fileLookup.getFileResolver().getPatternSetFactory();
                 }
             };
-            services.add(ScriptPluginFactory.class, DefaultScriptPluginFactory.this);
+            services.add(ScriptPluginApplicator.class, new DefaultScriptPluginApplicator(DefaultScriptPluginFactory.this, buildOperationExecutor));
```

Comment:
not really related to this change, but adding this service means that a `.gradle` script will only be able to apply other `.gradle` scripts, rather than a script of any type supported by Gradle.